### PR TITLE
feat: 🎸 Add triggerEventHandler method (#249)

### PR DESCRIPTION
**Pull request by Wykks**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/ngneat/spectator/issues/249

## What is the new behavior?

Add a new `triggerEventHandler ` method directly to spectator object.
Basically a helper on top of `debugElement.triggerEventHandler`, with type check output name/value when using component/directive symbol, and call detectChange.

Also support string selector in case you don't have access to the symbol.
(Don't know if this works with web component).

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

Not sure about the method name (because we have the dispatch prefix on other method, but this should be familiar since it's the name coming from angular testing), and the coding style. I'll add doc when all is okay :)

### CODE REVIEW

File: projects/spectator/src/lib/base/dom-spectator.ts [hunk 1]
```diff
@@ -175,6 +175,30 @@ export abstract class DomSpectator<I> extends BaseSpectator {
     return event;
   }
 
+  public triggerEventHandler<C = any, K extends keyof C = any>(
+    directiveOrSelector: Type<C> | string | DebugElement,
+    eventName: K,
+    eventObj: EventEmitterType<C[K]>
+  ) {
+    let debugElement: DebugElement;
```
Comment (NetanelBasal) [inline projects/spectator/src/lib/base/dom-spectator.ts]: Please extract it to a `getDebugElement` method and move it to our utils.

File: projects/spectator/src/lib/base/dom-spectator.ts [hunk 2]
```diff
@@ -175,6 +175,30 @@ export abstract class DomSpectator<I> extends BaseSpectator {
     return event;
   }
 
+  public triggerEventHandler<C = any, K extends keyof C = any>(
```
Comment (NetanelBasal) [inline projects/spectator/src/lib/base/dom-spectator.ts]: Nice to have: `K extends keyof C` - I think that we can create a type that'll extract only the keys that are of type `EventEmitter`.

File: projects/spectator/src/lib/base/dom-spectator.ts [hunk 3]
```diff
@@ -175,6 +175,30 @@ export abstract class DomSpectator<I> extends BaseSpectator {
...
+        // tslint:disable:no-console
+        console.error(`${directiveOrSelector} does not exists`);
+        return;
+      }
+    } else if (directiveOrSelector instanceof DebugElement) {
+      debugElement = directiveOrSelector;
+    } else {
+      debugElement = this.debugElement.query(By.directive(directiveOrSelector));
+    }
+    debugElement.triggerEventHandler(eventName as string, eventObj);
```
Comment (NetanelBasal) [inline projects/spectator/src/lib/base/dom-spectator.ts]: Add protection. `if (debugElement)`

File: projects/spectator/test/child-custom-event/child-custom-event-parent.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+describe('ChildCustomEventParentComponent', () => {
+  let spectator: Spectator<ChildCustomEventParentComponent>;
+  const createComponent = createComponentFactory({
+    component: ChildCustomEventParentComponent,
+    imports: [ChildCustomEventModule],
+    declareComponent: false
+  });
+
+  it('should trigger custom event with directive selector', () => {
```
Comment (NetanelBasal) [inline projects/spectator/test/child-custom-event/child-custom-event-parent.component.spec.ts]: Please update the docs.

File: projects/spectator/src/lib/base/dom-spectator.ts (hunk 2 above)
Comment (Wykks) [reply projects/spectator/src/lib/base/dom-spectator.ts]: I had to upgrade my TS knowledge but it's done 👍

File: projects/spectator/src/lib/base/dom-spectator.ts (hunk 1 above)
Comment (Wykks) [reply projects/spectator/src/lib/base/dom-spectator.ts]: Not sure where should I put this. There's a method `getNativeElement` at the bottom of this file. Should I create a `getDebugElement` next to it ?

Comment (NetanelBasal) [reply projects/spectator/src/lib/base/dom-spectator.ts]: Yes

File: projects/spectator/test/child-custom-event/child-custom-event-parent.component.spec.ts (hunk 4 above)
Comment (Wykks) [reply projects/spectator/test/child-custom-event/child-custom-event-parent.component.spec.ts]: Done
